semantic/graph.go

```diff
@@ -564,6 +565,25 @@ func (e *DictExpression) TypeOf() MonoType {
 	return e.Type
 }
 
+type VectorExpression struct {
+	Loc
+
+	Elements []Expression
+	Type     MonoType
+}
+
+func (e *VectorExpression) NodeType() string { return "VectorExpression" }
+
+func (e *VectorExpression) Copy() Node {
+	if e == nil {
+		return e
+	}
+
+	ne := new(VectorExpression)
+	*ne = *e
+	return ne
+}
+
```

Comment:
I don't think a vector expression will be needed for this work, even though there will be vector values.

Vectors are unlike arrays in this way. Users can create arrays with Flux syntax like `[1, 2, 3]` (that is an `ArrayExpression`) but users will never create literal vectors. Transformations like `map` will create records with vector fields, but that doesn't require vector expressions, since the fields can just be accesed via member expressions like `r.a` which will be typed as `vector of <sometype>`.

Does that make sense, or am I missing something?

Reply:
(correspondingly, I don't think a `VectorEvaluator` is needed either, unless I'm missing something)